index.js

```diff
@@ -1,88 +1,116 @@
 // ***** Question 1 *****
 // *** Uncomment the lines below to test
-// console.log("%cQuestion 1", "color: red") 
+console.log("%cQuestion 1", "color: red") 
 
-// console.log(counter) 
-// // => 2
-// console.log("%c----------", "color: red") 
+let counter = 1 
+counter = 2
+console.log(counter) 
+// => 2
+console.log("%c----------", "color: red") 
 
 
 // ***** Question 2 *****
 // *** Uncomment the lines below to test
 // *** after testing, comment the line causing the error back in 
 // *** otherwise, the error will stop the rest of your code from running
-// console.log("%cQuestion 2", "color: red") 
-
-// console.log(name)
-// => "Raffy"
-// name = "Not Raffy" 
-// => TypeError
-// console.log("%c----------", "color: red") 
+console.log("%cQuestion 2", "color: red") 
+const = Keith
+console.log(name)
+=> "Raffy"
+name = "Not Raffy" 
+=> TypeError
+console.log("%c----------", "color: red") 
 
 
 
 // ***** Question 3 *****
 // *** Uncomment the lines below to test
-// console.log("%cQuestion 3", "color: red") 
+console.log("%cQuestion 3", "color: red")
 
-// console.log(drinkWater(12))
-// // => "Man I sure am thirsty"
-// // => "Ahh that hits the spot"
-// // => 11
-// console.log("%c----------", "color: red") 
+function drink_water(current_thirst_level) {
+  console.log("Man I sure am thirsty")
+  current_thirst_level -= 1 
+  console.log("Ahh that hits the spot")
+  return current_thirst_level
+}
+console.log(drinkWater(12))
+// => "Man I sure am thirsty"
+// => "Ahh that hits the spot"
+// => 11
+console.log("%c----------", "color: red") 
 
 
 
 // ***** Question 4 *****
 // *** Uncomment the lines below to test
-// console.log("%cQuestion 4", "color: red") 
-
-// console.log(sameSameButDifferent(5, 5)) 
-// // => "same same"
+console.log("%cQuestion 4", "color: red") 
+
+function sameSameButDifferent(num, maybeNum) {
+  if num === maybeNum 
+    return "same same"
+  else if num == maybeNum 
+    return "same same (but different)"
+  else 
+  return "different"
+}
+console.log(sameSameButDifferent(5, 5)) 
```

Comment:
Close!

- It's standard to use parenthesis and curly braces for conditionals in Javascript. The lack of parenthesis around your conditional statements will cause an error. And curly braces around the conditional blocks increase your code's readability, makes your code more extensible, and reduces the risk of errors

```javascript
  if (num === maybeNum) { 
    return "same same"
}
  else if (num == maybeNum) { 
    return "same same (but different)"
}
  else {
  return "different"
}
```

- The key here was understanding the difference between `===` and `==`, so good job with that. You could also convert the string to a number with parseInt to avoid == :

`else if (num === parseInt(maybeNum))`